src/util.cpp

```diff
@@ -56,6 +56,40 @@ namespace rhost {
             return convert.to_bytes(ws);
         }
 #endif
+        // Effectively mbstowcs but also converts fancy quotes specified in the Unicode
+        // control range specifically, 0x91-0x94. These are 'Latin 1 supplement' characters
+        // and respectively are left/right single quotes and left/right double quotes.
+        // Example: https://everythingfonts.com/unicode/0x0091. Calling dQuote on a Latin
+        // string puts 0x93 and 0x94 quotes around the string and stock mbstowcs does not
+        // handle them correctly. Instead, it just makes 0x91 into U+0091 which is missing
+        // from most fonts since in Unicode 0x80-0xA0 does not contain visible characters.
+        size_t mbstowcs_withFancyQuotes(wchar_t *wc, char *s, size_t n) {
+            size_t nc = 0;
+            for (int i = 0, j = 0; nc < n && s[i] != '\0'; i++, j++) {
+                int char_size = mbtowc(wc + j, s + i, n - i);
+                if (char_size == 1) {
+                    switch (wc[j]) {
+                        case 0x0091:
+                            wc[j] = 0x2018; // Left single quote
+                            break;
+                        case 0x0092:
+                            wc[j] = 0x2019; // Right single quote
+                            break;
+                        case 0x0093:
+                            wc[j] = 0x201C; // Left double quote
+                            break;
+                        case 0x0094:
+                            wc[j] = 0x201D; // Right double quote
+                            break;
+                    }
+                }
+                i += char_size - 1;
```

Comment:
A bit hard to read. I think it would be simpler if you remove `i++` from loop definition and here it would be `i += char_size`